### PR TITLE
feat(core): dot sign transaction

### PR DESCRIPTION
**Pull request by gibrangul95**
@argjv @rliberal-simtlix @jrafael-simtlix @nimraarif21 @fdmirza

### CODE REVIEW

File: modules/core/test/v2/unit/coins/dot.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  let bitgo;
+  let basecoin;
+
+  before(function () {
+    bitgo = new TestBitGo({ env: 'mock' });
+    bitgo.initializeTestVars();
+    basecoin = bitgo.coin('tdot');
+  });
+
+  describe('Sign transaction', () => {
```
Comment (argjv) [inline modules/core/test/v2/unit/coins/dot.ts:14]: Can you add some error tests cases?

File: modules/core/src/v2/coins/dot.ts [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    }
+
+    if (!_.isNumber(addressVersion)) {
+      throw new Error('missing addressVersion parameter to sign transaction');
+    }
+
+    // if we are receiving addresses do not try to convert them
+    const signer = !accountLib.Dot.Utils.default.isValidAddress(params.txPrebuild.key)
+      ? new accountLib.Dot.KeyPair({ pub: params.txPrebuild.key }).getAddress()
+      : params.txPrebuild.key;
```
Comment (argjv) [inline modules/core/src/v2/coins/dot.ts:183]: You can switch the conditional validation to be non-negative (remove the `!`) and easier to read by swapping the if/else content

File: modules/core/src/v2/coins/dot.ts [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+    const self = this;
+    return co<SignedTransaction>(function* () {
+      const { txHex, signer, prv } = self.verifySignTransactionParams(params);
+      const factory = accountLib.register(self.getChain(), accountLib.Dot.TransactionBuilderFactory);
+      const txBuilder = factory.from(txHex);
+      txBuilder
+        .validity(params.txPrebuild.validity)
+        .referenceBlock(params.txPrebuild.referenceBlock)
+        .version(params.txPrebuild.version)
+        .sender({ address: signer })
```
Comment (argjv) [inline modules/core/src/v2/coins/dot.ts:209]: Are these parameters always provided? Or are they optional? If they are optional, we should validate they are present (and valid) before assigning them to the builder

File: modules/core/src/v2/coins/dot.ts [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+
+  parseTransaction(
+    params: ParseTransactionOptions,
+    callback?: NodeCallback<ParsedTransaction>
+  ): Bluebird<ParsedTransaction> {
+    return Bluebird.resolve({}).asCallback(callback);
+  }
+
+  verifyAddress(params: VerifyAddressOptions): boolean {
+    return true;
```
Comment (argjv) [inline modules/core/src/v2/coins/dot.ts:239]: Why not `accountLib.Dot.Utils.default.isValidAddress()`?

File: modules/core/src/v2/coins/dot.ts (hunk 3 above)
Comment (gibrangul95) [reply modules/core/src/v2/coins/dot.ts:209]: They're always mandatory. Wrote a test to ensure that.

File: modules/core/src/v2/coins/dot.ts (hunk 4 above)
Comment (argjv) [reply modules/core/src/v2/coins/dot.ts:239]: Or call `isValidAddress` from this class, no?

**Review by argjv — APPROVED** [review]

**Pull request merged**